src/haxelib/Data.hx

```diff
@@ -399,15 +350,22 @@ class Data {
 		}
 	}
 
-	/** Extracts project information from `jsondata`, validating it according to `check`. **/
-	public static function readData( jsondata: String, check : CheckLevel ) : Infos {
+	/**
+		Extracts project information from `jsondata`, validating it according to `check`.
+
+		`defaultName` is the project name to use if it is empty when the check value allows it.
+	**/
+	public static function readData( jsondata: String, check : CheckLevel, ?defaultName:ProjectName ) : Infos {
```

Comment:
Would `?defaultName = ProjectName.DEFAULT` work here?

Reply:
It says `Parameter default value should be constant`

Reply:
Oh right, never mind!